CFRunLoopGetMain() is actually in the Tiger Libraries, but not declared in the headers.
This fixes an error for an implicit definition of this function.
quartz.c: In function 'QuartzScreenSaver':
quartz.c:349: warning: implicit declaration of function 'CFRunLoopGetMain'
kencu@macports.org


diff --git a/hw/xquartz/quartz.c b/hw/xquartz/quartz.c
index 980aa4a36..ff873e219 100644
--- a/hw/xquartz/quartz.c
+++ b/hw/xquartz/quartz.c
@@ -78,6 +78,11 @@ extern OSErr UpdateSystemActivity(UInt8 activity);
 #define OverallAct 0
 #endif
 
+/* Work around a bug in Tiger's headers */
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1040 && MAC_OS_X_VERSION_MAX_ALLOWED < 1050
+extern "C" CFRunLoopRef CFRunLoopGetMain();
+#endif
+
 DevPrivateKeyRec quartzScreenKeyRec;
 int aquaMenuBarHeight = 0;
 QuartzModeProcsPtr quartzProcs = NULL;
